wallet/src/de/schildbach/wallet/ui/DecryptSeedWithPinDialog.kt

```diff
@@ -30,8 +29,11 @@ class DecryptSeedWithPinDialog : CheckPinDialog() {
         @JvmStatic
         fun show(activity: AppCompatActivity, requestCode: Int = 0, pinOnly: Boolean = false) {
             val checkPinDialog = DecryptSeedWithPinDialog()
-            if (PinRetryController.getInstance().isLocked) {
-                checkPinDialog.showLockedAlert(activity)
```

Comment:
Calling fragment methods from another object (a `companion object` in this case) is a bit dangerous. The fragment might not be attached and this might cause errors. In the past, we had an issue where in this method the `context` of the fragment was accessed instead of the passed context. I then had an issue with accessing the `viewModel` from that method.

I'm not sure if it's better to get strings beforehand and pass them to the method, or to move the dialog invocation out of there entirely. I've settled for the first option for now.